src/pyird/image/wavcal.py

```diff
@@ -1,11 +1,85 @@
+from pyird.io.read_linelist import read_linelist
 from astropy.io import fits
 import numpy as np
 from scipy.signal import medfilt
+from scipy.optimize import leastsq
 import pandas as pd
 
+import matplotlib.pyplot as plt
+
+def pdat2matrix(pdat):
+    data = np.zeros((l-j,2048))
+    m0 = j+1
+    for i in range(len(pdat)):
+        order, pixel, wav = pdat.iloc[i].values
+        order = order - m0
+        data[order,pixel] = wav
+    return data
+
+def fitfunc(XY,Ni,Nx,*params):
+    pix = XY[0][0]
+    m = XY[1][:,0]
+    minv = 1/XY[1]
+    p=[]
+    for p_tmp in params:
+        p.append(p_tmp)
+    p = np.array(p).reshape((Ni,Nx))
+    f = minv*np.polynomial.legendre.leggrid2d(m,pix,p)
+    return f.ravel()
+
+def errfunc(p,XY,data,Ni,Nx):
+    ind_data = data.ravel()!=0
+    data_on = data.ravel()[ind_data]
+    model = fitfunc(XY,Ni,Nx,p)
+    model_on = model[ind_data]
+    return data_on - model_on
+
+def fit_wav_solution(XY,data,Ni,Nx):
+    p0 = np.array([1]*Ni*Nx)
+    p1, cov = leastsq(errfunc,p0,args=((X,Y),data,Ni,Nx))
+    p1 = p1.reshape(Ni,Nx)
+    return p1
+
+def sigmaclip(data,wavsol,N=3):
+    zeroind = data.ravel()==0
+    residuals = (data.ravel()-wavsol.ravel())[~zeroind]
+    Nsigma = N*np.std(residuals)
+    n = 0
+    drop_ind = []
+    for i,data_tmp in enumerate((data.ravel())):
+        if data_tmp != 0:
+            res = data_tmp-wavsol[i]
+            if np.abs(res)>Nsigma:
+                #print(n, data_tmp)
+                drop_ind.append(n)
+            n += 1
+    return residuals,drop_ind
+
+def plot_result(wavsol,data):
+    wavsol_2d = wavsol.reshape(l-j,2048)
+    fig=plt.figure(figsize=(15,5))
+    ax1=fig.add_subplot(121)
+    for i in range(len(wavsol_2d[:,0])):
+        ax1.plot(wavsol_2d[i],color="tab:blue")
+        data_plot = np.copy(data[i])
+        data_plot[data_plot==0] = np.nan
+        ax1.plot(data_plot, ".",color="r")
+        #ax.text(2050, result_plot[i][-1]-5, i, color='green',fontsize=10)
+    ax1.set(xlabel="pixel",ylabel="wavelength [nm]")
+
+    ax2=fig.add_subplot(122)
+    for i,data_tmp in enumerate((data.ravel())):
+        if data_tmp != 0:
+            pix = i%2048 + 1
+            res = data_tmp-wavsol[i]
+            ax2.plot(pix,res,".",color="tab:blue")
+    ax2.set(xlabel="pixel",ylabel="residual [nm]")
+    plt.savefig('wavcal_final.png')
+    plt.show()
+
 if __name__ == '__main__':
     # Load a Th-Ar spectrum
-    tharfile = '/Users/kasagiyui/IRD/PhDwork/pyird/data/20210317/thar_mmf2_a_h.fits'
+    tharfile = '/Users/yuikasagi/IRD/PhDwork/pyird/data/20210317/thar_mmf2_a_h.fits'
```

Comment:
Can you put this file somewhere?
I wanna test the PR.

Reply:
I put it in 'pyird/data/thar_mmf2_a_H_20210317.fits'. Maybe it will work.

Reply:
I think this file should be external because of its size.

Reply:
Can you add the file name in MANIFEST.in like [this](https://github.com/prvjapan/pyird/blob/master/MANIFEST.in).
Then, you can call the file path by 
```python
import pkg_resources
path = (pkg_resources.resource_filename('pyird', 'data/thar_mmf2_a_H_20210317.fits'))
```